packages/core/core/src/types.ts

```diff
@@ -582,3 +582,38 @@ export interface Context {
 export interface FactoryParams {
   provide?: (context: Context) => any;
 }
+
+export interface ApiClientExtensionLifecycle {
+  beforeCreate?: (config, headers?: Record<string, string>) => any;
+  afterCreate?: ({ config, client }, headers?: Record<string, string>) => { config; client };
```

Comment:
why headers are optional? I guess they're always included, also - why we're not returning them? are they readonly? if so you can specify this in TS

also its better to have everything in 1 object for maintainability